UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/FunctionalTests/EnsSubscriptionRequestInQueueTest.cs

```diff
@@ -18,14 +18,18 @@ class EnsSubscriptionRequestInQueueTest
         private EnsApiClient EnsApiClient { get; set; }
         private TestConfiguration TestConfig { get; set; }
         private D365Payload D365Payload { get; set; }
-        private QueueClient Queue { get; set; }        
+        private QueueClient Queue { get; set; }
+        private string EnsToken { get; set; }
 
         [SetUp]
-        public void Setup()
+        public async Task SetupAsync()
         {
             TestConfig = new TestConfiguration();
             EnsApiClient = new EnsApiClient(TestConfig.EnsApiBaseUrl);
 
+            ADAuthTokenProvider adAuthTokenProvider = new();
+            EnsToken = await adAuthTokenProvider.GetEnsAuthToken();
+
             string filePath = Path.Combine(Directory.GetCurrentDirectory(), TestConfig.PayloadFolder, TestConfig.PayloadFileName);
 
             D365Payload = JsonConvert.DeserializeObject<D365Payload>(File.ReadAllText(filePath));
```

Comment:
As you've made the method `async` you can now use the `async` overload of `File.ReadAllText`:
```suggestion
            D365Payload = JsonConvert.DeserializeObject<D365Payload>(await File.ReadAllTextAsync(filePath));
```

Reply:
Done required changes.